File: backend/app/schemas/analytics.py
from typing import List, Optional
from pydantic import BaseModel, ConfigDict

class AnalyticsKpiCard(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    label: str
    value: str
    sub: str
    color: str

class HealthTrendPoint(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    period: str
    healthy: float
    atRisk: float

class DistributionSlice(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    name: str
    value: float
    color: str

class CategoryCount(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    name: str
    count: int

class GeographicHotspot(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    area: str
    emergencies: int
    atRisk: int
    trend: str  # 'up' | 'down' | 'stable'

class EnvironmentalInsight(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    icon: str
    title: str
    desc: str
    severity: str

class AnalyticsResponse(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    range: str
    range_label: str
    kpis: List[AnalyticsKpiCard]
    health_trend: List[HealthTrendPoint]
    current_distribution: List[DistributionSlice]
    emergency_categories: List[CategoryCount]
    hotspots: List[GeographicHotspot]
    insights: List[EnvironmentalInsight]

File: src/services/analytics.ts
import { apiRequest } from './api';

export type Range = '7d' | '30d' | '90d' | '1y';

export interface AnalyticsKpiCard {
  label: string;
  value: string;
  sub: string;
  color: string;
}

export interface HealthTrendPoint {
  period: string;
  healthy: number;
  atRisk: number;
}

export interface DistributionSlice {
  name: string;
  value: number;
  color: string;
}

export interface CategoryCount {
  name: string;
  count: number;
}

export interface GeographicHotspot {
  area: string;
  emergencies: number;
  atRisk: number;
  trend: 'up' | 'down' | 'stable' | string;
}

export interface EnvironmentalInsight {
  icon: string;
  title: string;
  desc: string;
  severity: string;
}

export interface AnalyticsResponse {
  range: string;
  range_label: string;
  kpis: AnalyticsKpiCard[];
  health_trend: HealthTrendPoint[];
  current_distribution: DistributionSlice[];
  emergency_categories: CategoryCount[];
  hotspots: GeographicHotspot[];
  insights: EnvironmentalInsight[];
}

export const analyticsApi = {
  getAnalytics: (range: Range = '90d') =>
    apiRequest<AnalyticsResponse>(`/api/analytics?range=${range}`, {
      method: 'GET',
    }),
};
